packages/design-system/src/components/Table/Table.jsx

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Alert from '../Alert/Alert';
+import PropTypes from 'prop-types';
+import TableCaption from './TableCaption';
+import classNames from 'classnames';
+
+function debounce(fn, ms) {
+  let timer;
+  return () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      timer = null;
+      fn.apply(this, arguments);
+    }, ms);
+  };
+}
+
+export const Table = ({
+  className,
+  responsiveTable,
+  striped,
+  scrollable,
+  scrollableActiveAlert,
+  children,
+  ...attributeOptions
+}) => {
+  const container = useRef(null);
+  // The captionID is calculated and stored as init value of a ref.
+  // This ensures that the ID remains constant for all renders.
+  const captionID = useRef('caption-' + Math.random().toString(36).substr(2, 9));
+  const [tabIndex, setTabIndex] = useState(null);
+
+  // Use the Effect hook to listen for window resize event.
+  // React execute useEffect() on each re-render and returns a function that cleans up the event listener from the render.
+  // https://reactjs.org/docs/hooks-effect.html
+  useEffect(() => {
+    // Set `tabIndex` to `0` if the table width is wider than the viewport and Table is set for horizontal scrolling.
+    if (scrollable) {
+      const { scrollWidth, clientWidth } = container.current;
+      const scrollableActive = scrollWidth > clientWidth;
+      setTabIndex(scrollableActive ? '0' : null);
+
+      const debouncedHandleResize = debounce(function handleResize() {
+        const { scrollWidth, clientWidth } = container.current;
+        const scrollableActive = scrollWidth > clientWidth;
+        setTabIndex(scrollableActive ? '0' : null);
+      }, 500);
+
+      // Create window resize event listener that calls the debouncedHandleResize function
+      window.addEventListener('resize', debouncedHandleResize);
+
+      // Specify how to clean up after this effect to avoid memory leak by removing establishd event listeners
+      return () => {
+        window.removeEventListener('resize', debouncedHandleResize);
+      };
+    }
+  });
+
+  const classes = classNames(
+    'ds-c-table',
+    striped ? 'ds-c-table--striped' : null,
+    responsiveTable ? `ds-c-table--stacked-${responsiveTable}` : null,
+    className
+  );
+
+  // `tabIndex` state is needed to make table container focusable and to display scrollable message when table width exceeds viewport.
+  // Set attribute `tabIndex = 0` to make table container focusable to enable keyboard support of using the arrow keys.
+  // Also, provide context for screen reader users as they are able to focus on the region.
+  // Do this by using table's <caption> to label the scrollable region using aira-labelleby
+  const attributescrollable = scrollable && {
+    className: 'ds-c-table__wrapper',
+    role: 'group',
```

Comment:
I did some research on the group role vs. region role. Some good information here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_group_role

I like keeping it as group because that means it doesn't surface programmatically to screen readers in navigation within the screen reader. Which is okay here because I don't think this info is that vital.